pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/MhloLegalizeUtils.cpp

```diff
@@ -655,5 +654,117 @@ Value getPermutedTensor(
       op->getLoc(), outTy, input, permuteAttr);
   return result.getResult();
 }
+
+Value getNormalizedDimSizeInternal(
+    PatternRewriter& rewriter,
+    Operation* op,
+    Value index,
+    Value dimSize) {
+  auto loc = op->getLoc();
+  Value zero = rewriter.create<arith::ConstantOp>(
+      loc, rewriter.getIntegerAttr(rewriter.getI64Type(), 0));
+  Value one = rewriter.create<arith::ConstantOp>(
+      loc, rewriter.getIntegerAttr(rewriter.getI64Type(), 1));
+
+  // To normalize index into range [-dimSize, dimSize]
+  // index = min(max(-dimSize, index), dimSize)
+  auto negDimSize = rewriter.create<arith::SubIOp>(loc, zero, dimSize);
+  index = rewriter.create<arith::MaxSIOp>(loc, negDimSize, index);
```

Comment:
Does priority is given to mhlo(mhlo.maximum) ?  cc @wyzero

Reply:
It's better to use `arith dialect` if we know it's shape computation. If not, we have to reply on some strategies to separate mhlo shape computation and mhlo data computation later, which may have to bad cases.